aee.js

```diff
@@ -82,3 +82,19 @@ Array.prototype.count = function (spec) {
   }
   return result
 }
+
+Array.prototype.max = function (comparer) {
+  const len = this.length
+  if (len === 0) return null
+  let result = this[0]
+  let comparerSpec = comparer
+  if (typeof comparerSpec !== 'function') {
+    comparerSpec = (!isNaN(this[0]) || this[0].constructor === Object)
```

Comment:
Just a comment / question on this line: if the element is a number or an object, it will use a comparer function in the `a - b` style, so it wont throw an error, and the default behaviour right now is that it returns the first object in the array, as logically the comparer is useless. The `this[0].constructor === Object` is the one that catches on if the elements of the array are complex objects instead of strings. Should I remove this and let it throw the default error (right now it would be `a.localeCompare is not a function`) or write my own exception?